src/NServiceBus.Core/Correlation/CorrelationContextExtensions.cs

```diff
@@ -10,6 +10,10 @@ public static class CorrelationContextExtensions
         /// </summary>
         /// <param name="options">Options being extended.</param>
         /// <param name="correlationId">The custom correlation id.</param>
+        [ObsoleteEx(
+            Message = "Setting a custom correlation ID is discouraged and will be removed in the next major version.",
```

Comment:
If the explanation isn't likely to fit into a single sentence, I'd prefer to point them to somewhere else, where we aren't constrained by length.